client/ayon_resolve/plugins/load/load_editorial_package.py

```diff
@@ -32,21 +36,126 @@ def load(self, context, name, namespace, data):
+
+        # get timeline media pool item for metadata update
+        timeline_media_pool_item = None
+        for item in lib.iter_all_media_pool_clips(root=loaded_bin):
+            item_name = item.GetName()
+            if item_name != loaded_timeline_name:
+                continue
+            timeline_media_pool_item = item
+            break
+
+        # Update the metadata
+        if timeline_media_pool_item:
+            clip_data = self._get_container_data(
+                context, data)
+
+            timeline_media_pool_item.SetMetadata(
+                lib.pype_tag_name, json.dumps(clip_data)
+            )
+
+            # set clip color based on random choice
+            clip_color = self.get_clip_color()
+            print("Clip color: ", clip_color)
+            timeline_media_pool_item.SetClipColor(clip_color)
+
+        # TODO: there are two ways to import timeline resources (representation
+        #   and resources folder) but Resolve seems to ignore any of this
+        #   since it is importing sources automatically. But we might need
+        #   to at least set some metadata to those loaded media pool items
         print("Timeline imported: ", timeline)
 
     def update(self, container, context):
-        # TODO: implement update method in future
-        pass
+        self.load(
+            context,
+            context["product"]["name"],
+            container["namespace"],
+            container
+        )
+
+    def _get_container_data(
+        self,
+        context: dict,
+        data: dict
+    ) -> dict:
+        """Return metadata related to the representation and version."""
+
+        # add additional metadata from the version to imprint AYON knob
+        version_entity = context["version"]
+
+        # remove unnecessary keys from the data
+        for key in ["_item", "name"]:
+            if key not in data:
+                continue
+            data.pop(key)
+
+        data = {
+            "load": data,
+        }
+
+        # add version attributes to the load data
+        data["load"].update(
+            version_entity["attrib"]
+        )
+
+        # add variables related to version context
+        data["load"].update(
+            {
+                "schema": "ayon:container-3.0",
+                "id": AVALON_CONTAINER_ID,
+                "loader": str(self.__class__.__name__),
+                "author": version_entity["data"]["author"],
+                "representation": context["representation"]["id"],
+                "version": version_entity["version"],
+            }
+        )
+
+        # add publish data for streamline publishing
+        data["publish"] = get_editorial_publish_data(
+            folder_path=context["folder"]["path"],
+            product_name=context["product"]["name"],
+            version=version_entity["version"],
+        )
+
+        return data
+
+    def get_clip_color(self):
+        """Return clip color based on version data."""
```

Comment:
Docstring is wrong. It doesn't base it on anything. Currently this is best of renamed to `get_random_clip_color()`

Reply:
done